### PR TITLE
Feat: Update starters

### PR DESCRIPTION
**Pull request by Meldiron**
## What does this PR do?

- Better copy in starters
- Use dynamic keys in starter

Runtimes to implement:
- [x] Bun
- [x] C++
- [x] Dart
- [x] Deno
- [x] Dotnet
- [x] Go
- [x] Java
- [x] Kotlin
- [x] Node
- [x] Node-typescript
- [x] PHP
- [x] Python
- [x] Python-ML
- [x] Ruby
- [x] Swift

## Test Plan

Manual QA flow:
- Deploy without scopes
- Try / (200)
- Try /ping (200)
- Check logs (error)
- Deploy with scopes
- Try / (200)
- Check logs (total users)

Runtimes to QA:

Each runtime tested with, and without scopes, and ping endpoint:

CleanShot 2024-08-05 at 13 14 28@2x

CleanShot 2024-08-05 at 13 14 46@2x

CleanShot 2024-08-05 at 17 32 16@2x

- [x] Bun
- [x] C++
- [x] Dart
- [x] Deno
- [x] Dotnet
- [x] Go
- [x] Java
- [x] Kotlin
- [x] Node
- [x] Node-typescript
- [x] PHP
- [x] Python
- [x] Python-ML
- [x] Ruby
- [x] Swift

## Related PRs and Issues

x

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes

### CODE REVIEW

**Review by eldadfux — APPROVED** [review]

File: bun/starter/src/main.ts [hunk 1]
```diff
@@ -1,29 +1,31 @@
...
+// This Appwrite function will be executed every time your function is triggered
 export default async ({ req, res, log, error }: any) => {
-  // Why not try the Appwrite SDK?
-  //
-  // const client = new Client()
-  //    .setEndpoint('https://cloud.appwrite.io/v1')
-  //    .setProject(Bun.env["APPWRITE_FUNCTION_PROJECT_ID"])
-  //    .setKey(Bun.env["APPWRITE_API_KEY"]);
+  // You can use the Appwrite SDK to interact with other services
+  // For this example, we're using the Auth service
```
Comment (christyjacob4) [inline bun/starter/src/main.ts:6]: Auth service? Or Users service ?

File: bun/starter/src/main.ts [hunk 2]
```diff
@@ -1,29 +1,31 @@
...
+  const client = new Client()
+    .setEndpoint(Bun.env["APPWRITE_FUNCTION_API_ENDPOINT"])
+    .setProject(Bun.env["APPWRITE_FUNCTION_PROJECT_ID"])
+    .setKey(req.headers['x-appwrite-key'] ?? '');
```
Comment (christyjacob4) [inline bun/starter/src/main.ts:10]: The user's API is supposed to work with a JWT or an API key right? 
What does `x-appwrite-key` contain? 

I believe we might need slightly different starters for server vs client SDKs?

File: bun/starter/src/main.ts [hunk 3]
```diff
@@ -1,29 +1,31 @@
...
-  // You can log messages to the console
-  log("Hello, Logs!");
-
-  // If something goes wrong, log an error
-  error("Hello, Errors!");
+  try {
+    const response = await users.list();
+    // Log messages and errors to the Appwrite Console
+    // These logs won't be seen by your end users
+    log(`Amount of users: ${response.total}`);
```
Comment (christyjacob4) [inline bun/starter/src/main.ts:17]:
```suggestion
    log(`Total users: ${response.total}`);
```

File: bun/starter/src/main.ts [hunk 4]
```diff
@@ -1,29 +1,31 @@
...
+  } catch(err) {
+    error("Could not list users: " + err.message);
+  }
 
   // The `req` object contains the request data
```
Comment (christyjacob4) [inline bun/starter/src/main.ts:22]:
```suggestion
  // The req object contains the request data
```
This isnt markdown so this doesn't really add any meaning here

File: bun/starter/src/main.ts [hunk 5]
```diff
@@ -1,29 +1,31 @@
...
-  if (req.method === "GET") {
-    // Send a response with the res object helpers
-    // `res.send()` dispatches a string back to the client
-    return res.send("Hello, World!");
+  if (req.path === "/ping") {
+    // Use `res` object to respond with `text()`, `json()`, or `binary()`
+    // Don't forget to return a response!
```
Comment (christyjacob4) [inline bun/starter/src/main.ts:25]: same here, lets skip the markdown

File: bun/starter/src/main.ts [hunk 6]
```diff
@@ -1,29 +1,31 @@
...
+    return res.text("Pong");
   }
 
-  // `res.json()` is a handy helper for sending JSON
```
Comment (christyjacob4) [inline bun/starter/src/main.ts:26]: any reason to remove this comment? It is still relevant

File: bun/starter/src/main.ts (hunk 2 above)
Comment (Meldiron) [reply bun/starter/src/main.ts:10]: x-appwrite-key is dynamic API key. This is starter for server

File: bun/starter/src/main.ts (hunk 1 above)
Comment (Meldiron) [reply bun/starter/src/main.ts:6]: users service ,updating

File: bun/starter/src/main.ts (hunk 6 above)
Comment (Meldiron) [reply bun/starter/src/main.ts:26]: It was mentioned earlier you can do json() on res. I removed this to keep the starter story (no double information).

Notice same for error() - we could again explain about logging, but we already explained in log()

**Review by christyjacob4 — APPROVED** [review]

**Comment by Meldiron** [comment]
Proof of QA:

CleanShot 2024-08-06 at 15 57 21@2x


CleanShot 2024-08-06 at 15 57 10@2x

File: ruby/starter/lib/main.rb [hunk 7]
```diff
@@ -1,36 +1,36 @@
...
-  context.log("Hello, Logs!")
-
-  # If something goes wrong, log an error
-  context.error("Hello, Errors!")
+  begin
+    response = users.list()
+    context.log("Total users: " + response.total.to_s)
+  rescue Exception => e
+    context.error("Could not list users: " + e.full_message)
+  end
```
Comment (christyjacob4) [inline ruby/starter/lib/main.rb:19]: missing the comment about `Log messages and errors to the Appwrite Console`

**Review by christyjacob4 — APPROVED** [review]

**Pull request merged**